pyaedt/edb_grpc/core/edb_data/padstacks_data.py

```diff
@@ -0,0 +1,1656 @@
+        """Parameters.
+
+        Returns
+        -------
+        list
+            List of parameters.
+        """
+        try:
+            pad_values = self._edb_padstack.GetData().GetPolygonalPadParameters(
+                self.layer_name, self.int_to_pad_type(self.pad_type)
+            )
+            return pad_values[1]
+        except:
+            return
+
+    @property
+    def parameters(self):
+        """Parameters.
+
+        Returns
+        -------
+        list
+            List of parameters.
+        """
+        pad_values = self._edb_padstack.GetData().GetPadParametersValue(
+            self.layer_name, self.int_to_pad_type(self.pad_type)
+        )
+
+        # pad_values = self._padstack_methods.GetPadParametersValue(self._edb_padstack, self.layer_name, self)
+        return [i.ToString() for i in pad_values[2]]
+
+    @parameters.setter
+    def parameters(self, propertylist):
+        if not isinstance(propertylist, list):
+            propertylist = [self._get_edb_value(propertylist)]
+        else:
+            propertylist = [self._get_edb_value(i) for i in propertylist]
+        self._update_pad_parameters_parameters(params=propertylist)
+
+    @property
+    def offset_x(self):
+        """Offset for the X axis.
+
+        Returns
+        -------
+        str
+            Offset for the X axis.
+        """
+
+        pad_values = self._edb_padstack.GetData().GetPadParametersValue(
+            self.layer_name, self.int_to_pad_type(self.pad_type)
+        )
+        return pad_values[3].ToString()
+
+    @offset_x.setter
+    def offset_x(self, offset_value):
+        self._update_pad_parameters_parameters(offsetx=offset_value)
+
+    @property
+    def offset_y(self):
+        """Offset for the Y axis.
+
+        Returns
+        -------
+        str
+            Offset for the Y axis.
+        """
+
+        pad_values = self._edb_padstack.GetData().GetPadParametersValue(
+            self.layer_name, self.int_to_pad_type(self.pad_type)
+        )
+        return pad_values[4].ToString()
+
+    @offset_y.setter
+    def offset_y(self, offset_value):
+        self._update_pad_parameters_parameters(offsety=offset_value)
+
+    @property
+    def rotation(self):
+        """Rotation.
+
+        Returns
+        -------
+        str
+            Value for the rotation.
+        """
+
+        pad_values = self._edb_padstack.GetData().GetPadParametersValue(
+            self.layer_name, self.int_to_pad_type(self.pad_type)
+        )
+        return pad_values[5].ToString()
+
+    @rotation.setter
+    def rotation(self, rotation_value):
+        self._update_pad_parameters_parameters(rotation=rotation_value)
+
+    @pyaedt_function_handler()
+    def int_to_pad_type(self, val=0):
+        """Convert an integer to an EDB.PadGeometryType.
+
+        Parameters
+        ----------
+        val : int
+
+        Returns
+        -------
+        object
+            EDB.PadType enumerator value.
```

Comment:
```suggestion
        """Convert an integer to an EDB pad geometry type.

        Parameters
        ----------
        val : int
            Value to convert.

        Returns
        -------
        object
            EDB.PadType enumerator value.
```
I'm not sure what the difference is between this one and the next one. I'm also finding GitHub's responsiveness to slow to continue editing.